cadc-uws-server/src/main/java/ca/nrc/cadc/uws/web/GetAction.java

```diff
@@ -0,0 +1,333 @@
+                }
+            }
+        } catch (JobNotFoundException ex) {
+            throw new ResourceNotFoundException("not found: " + jobID, ex);
+        } catch (JobPersistenceException ex) {
+            throw new RuntimeException("failed to access job pertsistence", ex);
+        } finally {
+            log.debug("DONE: " + syncInput.getPath());
+        }
+    }
+
+    private void writeJobListing() throws IOException, JobPersistenceException, TransientException {
+        Subject caller = AuthenticationUtil.getCurrentSubject();
+        AuthMethod am = AuthenticationUtil.getAuthMethod(caller);
+        if (am == null || AuthMethod.ANON.equals(am)) {
+            throw new AccessControlException("anonymous job listing not permitted");
+        }
+
+        List<String> phaseParams = syncInput.getParameters("PHASE");
+        List<ExecutionPhase> phaseList = new ArrayList<ExecutionPhase>();
+        if (phaseParams != null) {
+            for (String es : phaseParams) {
+                try {
+                    phaseList.add(ExecutionPhase.toValue(es));
+                } catch (IllegalArgumentException e) {
+                    throw new IllegalArgumentException("invalid UWS execution phase: PHASE=" + es);
+                }
+            }
+        }
+
+        String lastParam = syncInput.getParameter("LAST");
+        Integer lastInt = null;
+        if (lastParam != null) {
+            try {
+                lastInt = Integer.parseInt(lastParam);
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException("invalid integer: LAST=" + lastParam);
+            }
+
+            if (lastInt < 1) {
+                throw new IllegalArgumentException("invalid integer: LAST=" + lastParam);
+            }
+        }
+
+        String afterParam = syncInput.getParameter("AFTER");
+        Date afterDate = null;
+        if (afterParam != null) {
+            DateFormat df = DateUtil.getDateFormat(DateUtil.IVOA_DATE_FORMAT, DateUtil.UTC);
+            try {
+                afterDate = df.parse(afterParam);
+            } catch (ParseException e) {
+                throw new IllegalArgumentException("invalid IVOA timestamp: AFTER=" + afterParam);
+            }
+        }
+        String rp = syncInput.getRequestPath();
+        int i = rp.lastIndexOf("/");
+        rp = rp.substring(0, i);
+        log.debug("job requestPath to match: " + rp);
+        Iterator<JobRef> jobs = jobManager.iterator(rp, phaseList, afterDate, lastInt);
+        JobListWriter w = new JobListWriter();
+        syncOutput.setHeader("Content-Type", "text/xml");
+        OutputStream os = syncOutput.getOutputStream();
+        ByteCountOutputStream bc = new ByteCountOutputStream(os);
+        w.write(jobs, os);
+        logInfo.setBytes(bc.getByteCount());
+    }
+
+    private void writeJob(Job job) throws IOException {
+        // TODO: content negotiation via accept header
```

Comment:
Is the job authorization check done elsewhere?

Reply:
auth checks are in the JobManager so AccessControlException would be thrown from line 291